firmware/ubirch/m66/m66_http.c

```diff
@@ -83,26 +194,67 @@ size_t modem_http_read(uint8_t *buffer, uint32_t start, size_t size, uint32_t ti
   if(size <= 0) return 0;
 
   timer_set_timeout(timeout * 1000);
-  size_t available;
 
-  modem_send("AT+HTTPREAD=%d,%d", start, size);
-  modem_expect_scan("+HTTPREAD: %lu", timeout, &available);
+  int file_handle = 0;
+  size_t  data_len = 0;
+  bool close_file = false;
 
-  size_t idx = modem_read_binary(buffer, available, uTimer_Remaining);
-  if (!modem_expect_OK(uTimer_Remaining)) return 0;
+  file_handle = modem_http_file_open(file_name, 0, uTimer_Remaining);
+
+  if (!file_handle) return 0;
 
-  CIODUMP(buffer, idx);
+  /*
+   *  Position -> Pointer movement mode
+   *  0 -> File begining | Default value
+   *  1 -> Current position of the pointer
+   *  2 -> File end
+   */
+  modem_send("AT+QFSEEK=%d,%d,%d", file_handle, start, 0);
+  if (!modem_expect_OK(uTimer_Remaining))
+  {
+    CSTDEBUG("HTTP --- Failed ot seek the file position\r\n");
+    return 0;
+  }
 
-  return idx;
+  data_len = modem_http_file_read(buffer, file_handle, size, uTimer_Remaining);
+  if (!data_len) return 0;
+
+  close_file = modem_http_file_close(file_handle, uTimer_Remaining);
+  if (!close_file) return 0;
+
+  return data_len;
 }
 
-int modem_http_get(const char *url, size_t *res_size, uint32_t timeout) {
+int modem_http_get(const char *url, size_t *res_size, uint32_t timeout)
+{
   timer_set_timeout(timeout * 1000);
 
-  int status = modem_http_prepare(url, timeout);
-  if (status) return status;
+  int dl_size = 0;
+
+  modem_send("AT+QHTTPURL=%d,%d", strlen(url), timer_timeout_remaining() / 1000);
+
+  if (!modem_expect("CONNECT", uTimer_Remaining))
+  {
+    CSTDEBUG("HTTP --- Failed to receive CONNECT\r\n");
+    return false;
+  }
+
+  modem_send(url);
+
+  if (!modem_expect_OK(uTimer_Remaining))
+  {
+    CSTDEBUG("HTTP --- Failed to connect to the server");
+    return 0;
+  }
 
-  return modem_http(HTTP_GET, res_size, uTimer_Remaining);
+  modem_send("AT+QHTTPGET=%d", timer_timeout_remaining() / 1000);
+  if (!modem_expect_OK(uTimer_Remaining)) return 0;
+
+  dl_size = modem_http_file_dl(file_name, uTimer_Remaining);
+
+  if (!dl_size) return 0;
+
+  return dl_size;
```

Comment:
I don't think you should change the API return value here, the response size should be stored in the resize check that please.